src/lib.rs

```diff
@@ -35,10 +37,23 @@ impl KVMi {
                           unsafe extern fn(*const kvmi_qemu2introspector,
                                            *mut kvmi_introspector2qemu,
                                            *mut c_void) -> c_int);
-        let ctx = null_mut();
-        unsafe {
-            kvmi_sys::kvmi_init_unix_socket(socket_path.as_ptr(), accept_db, hsk_cb, ctx);
+        let mut kvmi = KVMi {
+            ctx: null_mut(),
+            connect_condvar: Condvar::new(),
+        };
+        let cb_ctx: *mut c_void = &kvmi;
+        // kvmi_dom = NULL
+        let guard = Mutex::new(false);
+        let lock = guard.lock().unwrap();
+        let res: *mut c_void = unsafe {
+            kvmi_sys::kvmi_init_unix_socket(socket_path.as_ptr(), accept_db, hsk_cb, cb_ctx)
```

Comment:
I trying this:
~~~Rust
kvmi_sys::kvmi_init_unix_socket(socket_path.as_ptr(), accept_db, hsk_cb, &kvmi_con as _)
~~~

It doesn't work:
casting `&KVMiCon` as `*mut std::ffi::c_void` is invalid

Reply:
Hi @Wenzel , you may try this

```rust
&kvmi_con as *mut KVMiCon as _
```
or even more succint

```rust
&kvmi_con as *mut _ as _
```

because any `&mut T` can be implicitly coerced into `*mut T`.

Reply:
Also tried what you suggested:
~~~Rust
kvmi_sys::kvmi_init_unix_socket(socket_path.as_ptr(), accept_db, hsk_cb, &kvmi_con as *mut _ as _)
~~~

doesn't work for the compiler:
`error[E0606]: casting `&KVMiCon` as `*mut _` is invalid`

Reply:
Sorry

```rust
&mut kvmi_con as *mut _ as _
```